graylog2-server/src/main/java/org/graylog2/rest/resources/system/ClusterResource.java

```diff
@@ -53,14 +67,39 @@
 @Produces(MediaType.APPLICATION_JSON)
 public class ClusterResource extends RestResource {
     private final NodeService nodeService;
+    private final PaginatedNodeService paginatedNodeService;
```

Comment:
When I see the dependencies here, I'd be thinking about moving it to an own resource.